README.md

```diff
@@ -2,4 +2,114 @@
+
+
+## Usage
+
+The core component of the SDK is the #[marine] macro that should be used with export functions, extern blocks, and structures. Let's consider a simplest scenario with a module with one export function:
+```rust
+#[marine]
+use marine_rs_sdk::marine;
+use marine_rs_sdk::module_manifest;
+
+#[marine]
+pub fn greeting(name: String) -> String {
+    format!("Hi, {}", name)
+}
+```
+This code imports the Marine SDK and wraps the greeting function with the #[marine] macro. Every function wrapped in such way will be exported from a compiled Wasm module.
+
+To build this module into Wasm, you'll be needed the marine CLI tool (`cargo install marine` to install):
+
+```bash
+marine build --release
+```
+
+Finally, you could obtain some build info from the resulted binary and play with it with the help of mrepl (`cargo install mrepl` to install):
+
+```bash
+> marine info ./target/wasm32-wasi/release/greeting.wasm
+it version:  0.23.1
+sdk version: 0.6.0
+authors:     <user-name>
+version:     0.1.0
+description:
+repository:
+build time:  2022-05-15 14:52:37.865550 +00:00 UTC
+
+> mrepl --quiet
+1> load greeting ./target/wasm32-wasi/release/greeting.wasm
+module successfully loaded into App service
+elapsed time: 52.153308ms
+
+2> interface
+Application service interface:
+greeting:
+  fn greeting(name: String) -> String
+
+3> call greeting greeting "user"
+result: String("Hi, user")
+ elapsed time: 132.021µs
+```
+
+The complete guide of developing this simple module could be found [here](https://fluence.dev/docs/marine-book/quick-start/develop-a-single-module-service).
+
+
+## SDK components
+
+The SDK exports the following major components:
+- `#[marine]` procedure macro that could be applied to
+    - [function](https://fluence.dev/docs/marine-book/marine-rust-sdk/developing/export-functions) making it export from a module
+    - [extern block](https://fluence.dev/docs/marine-book/marine-rust-sdk/developing/import-functions) making it imported from a module in a Marine-suitable way
+    - [structure](https://fluence.dev/docs/marine-book/marine-rust-sdk/developing/structures), making it usable as an argument of an export or an import function
+- [call parameters interface](https://fluence.dev/docs/marine-book/marine-rust-sdk/developing/call-parameters) intended to provide a set of module start parameters
+- [mounted binaries interface](https://fluence.dev/docs/marine-book/marine-rust-sdk/developing/mounted-binaries) that can be used to call a CLI tool
+- [module_manifest](https://fluence.dev/docs/marine-book/marine-rust-sdk/developing/module-manifest) macro intended to embed some info into a compiled Wasm
+- [logging stuff](https://fluence.dev/docs/marine-book/marine-rust-sdk/developing/logging) that allows adjusting a Marine logging mechanism
+
+
+## Supported Rust types
+
+At the time, the `#[marine]` macro allows using the following Rust types as an argument of export and import function or field of a structure:
```

Comment:
Perhaps, "At the time of this writing," or "At the moment," instead of "At the time," but I am not 100% sure.